File: tools/improve/tests/original/maudec/cpp/collatz.cpp
#include <optional>


std::optional<int> rlapp_Nat(int term) {
	if (term % 2 == 0) {
		return std::make_optional(term / 2);
	}
	if (term % 2 == 1) {
		return std::make_optional(1 + term * 3);
	}
	return std::nullopt;
}

std::optional<int> arlapp_Nat(int term) {
	return rlapp_Nat(term);
}

File: tools/improve/tests/gemini-3-flash/maudec/rust/collatz.rs
//
//	<comment from="gemini-3-flash-preview">
//	The implementation was simplified by matching on the result of `term % 2` directly. In Rust, the remainder operator `%` returns 0 for even numbers, 1 for positive odd numbers, and -1 for negative odd numbers. By using this approach, we avoid redundant calculations and clarify the logic for handling each integer parity. The `arlapp_Nat` function remains a transparent wrapper to maintain the requested signatures.
//	</comment>
//

fn rlapp_Nat(term: i32) -> Option<i32> {
    match term % 2 {
        0 => Some(term / 2),
        1 => Some(term * 3 + 1),
        _ => None,
    }
}

fn arlapp_Nat(term: i32) -> Option<i32> {
    rlapp_Nat(term)
}